### PR TITLE
KAFKA-3845: KIP-75: Add per-connector converters

### PR DESCRIPTION
**Pull request by ewencp**

### CODE REVIEW

**Comment by ewencp** [comment]
@hachikuji @Ishiihara @shikhar @ijuma @gwenshap

**Comment by ewencp** [comment]
And now with passing system tests: https://jenkins.confluent.io/job/system-test-kafka-branch-builder-2/79

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java [hunk 1]
```diff
@@ -64,7 +72,9 @@ public static ConfigDef configDef() {
         return new ConfigDef()
             .define(NAME_CONFIG, Type.STRING, Importance.HIGH, NAME_DOC, COMMON_GROUP, 1, Width.MEDIUM, NAME_DISPLAY)
             .define(CONNECTOR_CLASS_CONFIG, Type.STRING, Importance.HIGH, CONNECTOR_CLASS_DOC, COMMON_GROUP, 2, Width.LONG, CONNECTOR_CLASS_DISPLAY)
-            .define(TASKS_MAX_CONFIG, Type.INT, TASKS_MAX_DEFAULT,  atLeast(TASKS_MIN_CONFIG), Importance.HIGH, TASKS_MAX_DOC, COMMON_GROUP, 3, Width.SHORT, TASK_MAX_DISPLAY);
+            .define(TASKS_MAX_CONFIG, Type.INT, TASKS_MAX_DEFAULT, atLeast(TASKS_MIN_CONFIG), Importance.HIGH, TASKS_MAX_DOC, COMMON_GROUP, 3, Width.SHORT, TASK_MAX_DISPLAY)
+            .define(KEY_CONVERTER_CLASS_CONFIG, Type.CLASS, null, Importance.LOW, KEY_CONVERTER_CLASS_DOC, COMMON_GROUP, 4, Width.SHORT, KEY_CONVERTER_CLASS_DISPLAY)
+            .define(VALUE_CONVERTER_CLASS_CONFIG, Type.CLASS, null, Importance.LOW, VALUE_CONVERTER_CLASS_DOC, COMMON_GROUP, 4, Width.SHORT, VALUE_CONVERTER_CLASS_DISPLAY);
```
Comment (shikhar) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java]: `orderInGroup` param is duplicated for key & value converter

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerder.java [hunk 2]
```diff
@@ -206,14 +206,17 @@ public synchronized void restartTask(ConnectorTaskId taskId, Callback<Void> cb)
         if (!configState.contains(taskId.connector()))
             cb.onCompletion(new NotFoundException("Connector " + taskId.connector() + " not found", null), null);
 
-        Map<String, String> taskConfig = configState.taskConfig(taskId);
-        if (taskConfig == null)
+        Map<String, String> taskConfigs = configState.taskConfig(taskId);
```
Comment (shikhar) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerder.java]: would `taskConfigParams` or  be a better name as those get used to create the `taskConfig` object?

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerder.java [hunk 3]
```diff
@@ -206,14 +206,17 @@ public synchronized void restartTask(ConnectorTaskId taskId, Callback<Void> cb)
...
+        if (taskConfigs == null)
             cb.onCompletion(new NotFoundException("Task " + taskId + " not found", null), null);
+        TaskConfig taskConfig = new TaskConfig(taskConfigs);
+        Map<String, String> connConfigs = configState.connectorConfig(taskId.connector());
```
Comment (shikhar) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerder.java]: as above, though this one seems inlinable

**Comment by shikhar** [comment]
LGTM other than minor comments

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java (hunk 1 above)
Comment (ewencp) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java]: Ack, good catch

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerder.java (hunk 3 above)
Comment (ewencp) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerder.java]: did rename + cleanup here and in DistributedHerder

**Comment by gwenshap** [comment]
LGTM.

**Pull request closed without merging**